### ISSUE TITLE
Forbid duplicate test names

### ISSUE THREAD

**Issue by lo1tuma** [lo1tuma/eslint-plugin-mocha#33]

**Comment by lo1tuma** [comment]
The following patterns should be considered a warning 

```js
it('foo');
it('foo');
```

We need to decide if we want to only disallow duplicates in the same `describe` level or in the whole file. 
Should we also disallow duplicate describe names?

**Comment by jfmengels** [comment]
> We need to decide if we want to only disallow duplicates in the same describe level or in the whole file.

In the same describe level.
```js
describe('foo', function() {
  it('should throw when called with null', function() {})
});
describe('bar', function() {
  it('should throw when called with null', function() {})
});
```

I think it's fine to have the same title here, as the end title would be different for both.

> Should we also disallow duplicate describe names?

Yes. Ideally, this should be project wide for top-level describe, but ESLint is not fit to check that.
